File: java/Persona1.java
public class Persona1 {
    // 1. DATI PRIVATI: Nessuno dall'esterno può scrivere persona.nome = "..."
    private String nome;
    private String cognome;

    Persona1(String nome, String cognome){
        this.nome = nome;
        this.cognome = cognome;
    }

    // 2. GETTER: Metodo pubblico per "esporre" il valore in lettura
    public String getNome() {
        return nome;
    }

    // 3. SETTER: Metodo pubblico per permettere la modifica controllata
    public void setNome(String nome) {
        // Qui potresti aggiungere controlli (es. if(nome != null))
        this.nome = nome;
    }

    public String getCognome() {
        return cognome;
    }

    public void setCognome(String cognome) {
        this.cognome = cognome;
    }

    // IL METODO COPY: Prende i valori da un'altra persona e li scrive su di sé.
    // Questo crea una copia REALE dei dati, non del puntatore.
    public void copy(Persona1 personaDaCopiare) {
        this.setNome(personaDaCopiare.getNome());
        this.setCognome(personaDaCopiare.getCognome());
    }

    void saluta() {
        System.out.println("Ciao, sono una persona generica.");
    }

}

File: C#/Persona1.cs
using System;


namespace Lezione29Class{
    public class Persona1{
        // Campi privati (Incapsulamento)
        private string nome;
        public string cognome;
        public int eta;
        // ==========================================
        // APPROCCIO 1: METODI GETTER E SETTER (Stile classico)
        // ==========================================
        public void SetNome(string nome){
            this.nome = nome;
        }
        public void SetCognome(string cognome){
            this.cognome = cognome;
        }
        public void SetEta(int eta){
            if(eta >= 0) this.eta = eta;
            else Console.WriteLine("Hai messo un eta negativa, imperdonabile!!");
        }
        public string GetNome(){
            return this.nome;
        }

        public string GetCognome(){
            return this.cognome;
        }
        public int GetEta(){
            return this.eta;
        }
        // ==========================================
        // APPROCCIO 2: LE PROPRIETÀ DI C# (Modo corretto e nativo)
        // ==========================================
        public string Nome {
            get => this.nome;
            set => this.nome = value; // 'value' è la parola chiave che contiene il dato in ingresso
        }

        // ==========================================
        // COSTRUTTORI
        // ==========================================
        // Costruttore completo
        public Persona1(string nome, string cognome, int eta) {
            this.nome = nome;
            this.cognome = cognome;
            this.eta = eta;
        }
        public Persona1() { }
        public void saluta() {
            Console.WriteLine($"Ciao sono {nome}");
        }
    }
}